src/renderer/components/swap/Swap.tsx

```diff
@@ -390,6 +389,156 @@ export const Swap = ({
     )
   }, [assetsToSwap, onConfirmSwap, changeAmount, closePrivateModal])
 
+  const sourceChainFee: RD.RemoteData<Error, BaseAmount> = useMemo(
+    () =>
+      FP.pipe(
+        feesProp,
+        RD.map(({ source }) => source)
+      ),
+    [feesProp]
+  )
+
+  const targetChainFee: RD.RemoteData<Error, BaseAmount> = useMemo(
+    () =>
+      FP.pipe(
+        feesProp,
+        RD.map(({ target }) => target)
+      ),
+    [feesProp]
+  )
+
+  const sourceChainBalanceError: boolean = useMemo(
+    () =>
+      FP.pipe(
+        sequenceTOption(sourceAsset, walletBalances),
+        O.map(([sourceAsset, walletBalances]) => {
+          // If sourceAsset is the same as source chain
+          // we should subtract changeAmount
+          const shouldSubtractChangeAmount = eqAsset.equals(sourceAsset, getChainAsset(sourceAsset.chain))
+
+          const balanceMinusAmount = FP.pipe(
+            walletBalances,
+            A.findFirst((walletBalance) => eqAsset.equals(walletBalance.asset, sourceAsset)),
+            O.map((sourceBalance) =>
+              baseToAsset(sourceBalance.amount)
+                .amount()
+                .minus(shouldSubtractChangeAmount ? changeAmount : ZERO_BN)
+            ),
+            O.map(FP.flow(assetAmount, assetToBase))
+          )
+
+          return FP.pipe(
+            sequenceTOption(balanceMinusAmount, FP.pipe(sourceChainFee, RD.toOption)),
+            O.map(([leftBalance, sourceChainFee]) => {
+              const sourceChainFeeAmount = sourceChainFee.amount()
+
+              return leftBalance.amount().minus(sourceChainFeeAmount)
+            }),
+            O.map((result) => result.isNegative()),
+            O.getOrElse((): boolean => false)
+          )
+        }),
+        O.getOrElse((): boolean => false)
+      ),
+    [changeAmount, sourceAsset, walletBalances, sourceChainFee]
+  )
+
+  const sourceChainErrorLabel: JSX.Element = useMemo(() => {
+    if (!sourceChainBalanceError) {
+      return <></>
+    }
+    const sourceChainAsset = FP.pipe(
+      sourceAsset,
+      O.map(({ chain }) => getChainAsset(chain)),
+      O.map((asset) => asset.symbol),
+      O.getOrElse(() => '')
+    )
+
+    return (
+      <Styled.BalanceErrorLabel>
+        {intl.formatMessage({ id: 'swap.errors.amount.balanceShouldCoverChainFee' }, { asset: sourceChainAsset })}{' '}
+      </Styled.BalanceErrorLabel>
+    )
+  }, [sourceChainBalanceError, sourceAsset, intl])
+
+  const targetChainFeeAmountInTargetAsset: BaseAmount = useMemo(() => {
+    const chainFee = FP.pipe(
+      targetChainFee,
+      RD.getOrElse(() => ZERO_BASE_AMOUNT)
+    )
+
+    return FP.pipe(
+      targetAsset,
+      O.map((asset) => {
+        const chainAsset = getChainAsset(asset.chain)
+        return getValueOfAsset1InAsset2(chainFee, poolData[assetToString(chainAsset)], poolData[assetToString(asset)])
+      }),
```

Comment:
Another note:  Whenever asset === asset.chain you can't use `getValueOfAsset1InAsset2`. In this case just use [chainFee](https://github.com/thorchain/asgardex-electron/pull/663/files#diff-35dddf5143f357d65494359d2b087faeca6e8eb13706b5a652f73f5dfe3a86eeR465) - see [calculateAssetChainFee in Withdraw](https://github.com/thorchain/asgardex-electron/pull/662/files#diff-9523733ca0dcea33995c77fb2a65e28b89a9873ae0e9443005f3180c08fc2f5eR140-R146)